protocol/x/clob/e2e/short_term_orders_test.go

```diff
@@ -1020,3 +1019,177 @@ func TestShortTermOrderReplacements(t *testing.T) {
 		})
 	}
 }
+
+func TestCancelShortTermOrder(t *testing.T) {
+	tests := map[string]struct {
+		firstBlockOrders   []clobtypes.MsgPlaceOrder
+		firstBlockCancels  []clobtypes.MsgCancelOrder
+		secondBlockOrders  []clobtypes.MsgPlaceOrder
+		secondBlockCancels []clobtypes.MsgCancelOrder
+
+		expectedOrderIdsInMemclob map[clobtypes.OrderId]bool
+		expectedCancelsInMemclob  map[clobtypes.OrderId]bool
+		expectedOrderFillAmounts  map[clobtypes.OrderId]uint64
+	}{
+		"Cancel unfilled short term order": {
```

Comment:
can we get one to test `existing short-term cancel has the same or greater `goodTilBlock``? we shouldn't place the new cancel then